lib/types/db.ts

```diff
@@ -0,0 +1,53 @@
+import Sequelize, { DataTypeAbstract, DefineAttributeColumnOptions } from 'sequelize';
+import { db } from './index';
+
+export type SequelizeAttributes<T extends { [key: string]: any }> = {
+  [P in keyof T]: string | DataTypeAbstract | DefineAttributeColumnOptions
+};
+
+export type CurrencyFactory = {
+  id: string;
+};
+
+export type CurrencyAttributes = CurrencyFactory;
+
+export type CurrencyInstance = CurrencyAttributes & Sequelize.Instance<CurrencyAttributes>;
```

Comment:
I'm a little unclear as to how `Sequelize.Instance<CurrencyAttributes>` is being used here, but the result is that all the `Instance` types in this class are treated as `any`. I can assign a number value to `CurrencyInstance`, for example, and my compiler accepts it. So something is clearly off here.

Reply:
Gotcha. I realized my problem was that I didn't run `npm install` after checking out this branch and therefore didn't have the `Sequelize` types available, which is why I was getting `any` here. So you can ignore this.